modules/400-descheduler/hooks/internal/api/v1alpha1/descheduler.go

```diff
@@ -0,0 +1,272 @@
+}
+
+type ResourcesRequests struct {
+	// Defines VPA mode that applies to a given descheduler Deployment
+	// +kubebuilder:default=VPA
+	// +kubebuilder:validation:Enum=Manual;VPA;Off
+	Mode ResourcesRequestsMode `json:"mode,omitempty"`
+}
+
+type ResourcesRequestsMode string
+
+const (
+	ResourcesRequestsModeManual = "Manual"
+	ResourcesRequestsModeVPA    = "VPA"
+	ResourcesRequestsModeOff    = "Off"
+)
+
+type DeschedulerPolicy struct {
+	// Parameters that apply to all policies
+	CommonParameters CommonParameters `json:"parameters,omitempty"`
+
+	// List of strategies with corresponding parameters for a given Descheduler instances
+	// To enable a strategy without giving it any parameters, specify it like this:
+	// removePodsViolatingNodeAffinity: {}
+	Strategies DeschedulerStrategies `json:"strategies,omitempty"`
+}
+
+type CommonParameters struct {
+	// NodeSelector for a set of nodes to operate over
+	NodeSelector *string `json:"nodeSelector,omitempty"`
+
+	// EvictFailedBarePods allows pods without ownerReferences and in failed phase to be evicted.
+	EvictFailedBarePods *bool `json:"evictFailedBarePods,omitempty"`
+
+	// EvictLocalStoragePods allows pods using local storage to be evicted.
+	EvictLocalStoragePods *bool `json:"evictLocalStoragePods,omitempty"`
+
+	// EvictSystemCriticalPods allows eviction of pods of any priority (including Kubernetes system pods)
+	// Note: Setting evictSystemCriticalPods to true disables priority filtering entirely.
+	EvictSystemCriticalPods *bool `json:"evictSystemCriticalPods,omitempty"`
+
+	// IgnorePVCPods prevents pods with PVCs from being evicted.
+	IgnorePVCPods *bool `json:"ignorePvcPods,omitempty"`
+
+	// MaxNoOfPodsToEvictPerNode restricts maximum of pods to be evicted per node.
+	MaxNoOfPodsToEvictPerNode *int `json:"maxNoOfPodsToEvictPerNode,omitempty"`
+
+	// MaxNoOfPodsToEvictPerNamespace restricts maximum of pods to be evicted per namespace.
+	MaxNoOfPodsToEvictPerNamespace *int `json:"maxNoOfPodsToEvictPerNamespace,omitempty"`
+}
+
+type DeschedulerStrategies struct {
+	RemoveDuplicates *RemoveDuplicates `json:"removeDuplicates,omitempty"`
+
+	LowNodeUtilization *LowNodeUtilization `json:"lowNodeUtilization,omitempty"`
+
+	HighNodeUtilization *HighNodeUtilization `json:"highNodeUtilization,omitempty"`
+
+	RemovePodsViolatingInterPodAntiAffinity *RemovePodsViolatingInterPodAntiAffinity `json:"removePodsViolatingInterPodAntiAffinity,omitempty"`
+
+	RemovePodsViolatingNodeAffinity *RemovePodsViolatingNodeAffinity `json:"removePodsViolatingNodeAffinity,omitempty"`
+
+	RemovePodsViolatingNodeTaints *RemovePodsViolatingNodeTaints `json:"removePodsViolatingNodeTaints,omitempty"`
+
+	RemovePodsViolatingTopologySpreadConstraint *RemovePodsViolatingTopologySpreadConstraint `json:"removePodsViolatingTopologySpreadConstraint,omitempty"`
+
+	RemovePodsHavingTooManyRestarts *RemovePodsHavingTooManyRestarts `json:"removePodsHavingTooManyRestarts,omitempty"`
```

Comment:
I don't think we want to allow modifying every option. Or, there should be some reasonable defaults in the scheme (to make these defaults explicit).

Reply:
Done the defaulting in openapi.

Reply:
Reasonable defaults are now present in openapi.